### DISCUSSION TITLE
How to scroll to specific coordinates

### DISCUSSION THREAD

**Discussion by maryan-mandzyuk** [tradingview/lightweight-charts#1194] [Q&A]
I need to scroll to the time when the trade operation happened I have the time of trade convert it to coordinate and then use scrollToPosition function but it’s not working. Could you help me? How can I scroll to position when I have time?

```
const coordinate = timeScale.timeToCoordinate(new Date(someTime).getTime() / 1000);
timeScale.scrollToPosition(coordinate, true)
```

**Comment by SlicedSilver** [comment]
The `scrollToPosition` method scrolls to a position relative to the last bar on the chart.

So `scrollToPosition(0)` will scroll the chart such that the last point is right on the right edge of the chart. Where as, `scrollToPosition(-10)` would scroll the chart such that the last 10 bars are past the right edge of the chart.

To scroll to a specific date, you would need to know that data points index relative to the last point. So if you have a 100 data points and you want to scroll to point 75, you would put `scrollToPosition(-25)`.

You could also use `setVisibleRange` to move the chart to a specific date. For example, if you want to show 40 bars of data then you could do something like this: `setVisibleRange({from: myTargetDate - 20 * days, to: myTargetDate + 20 * days` where `days = 24 * 60 * 60` seconds.

**Reply by jy0000** [reply]
Is there no way to have it scroll relative to another position. Or the first bar on the chart. Because I have a chart with data constantly being updated, I want it to be able to scroll back relative to the current time it is displaying

**Reply by SlicedSilver** [reply]
The timeScale API ([`ITimeScaleApi`](https://tradingview.github.io/lightweight-charts/docs/api/interfaces/ITimeScaleApi)) has a few methods for converting between indices, coordinates and times. If you know the time you want to scroll to a specific timestamp then you can do the following:
```js
function scrollToTime(time, timeScale){
    const currentPosition = timeScale.scrollPosition();
    const currentVisibleLogicalRange = timeScale.getVisibleLogicalRange();

    const coordinate = timeScale.timeToCoordinate(time);
    const logicalIndex = timeScale.coordinateToLogical(coordinate);

    const targetPosition = currentPosition - currentVisibleLogicalRange.to + logicalIndex;
    timeScale.scrollToPosition(targetPosition, true);
}
```

- [timeToCoordinate](https://tradingview.github.io/lightweight-charts/docs/api/interfaces/ITimeScaleApi#timetocoordinate)
- [coordinateToLogical](https://tradingview.github.io/lightweight-charts/docs/api/interfaces/ITimeScaleApi#coordinatetological)
- [scrollToPosition](https://tradingview.github.io/lightweight-charts/docs/api/interfaces/ITimeScaleApi#scrolltoposition)
